templates/workloads/partials/cf/subscribe.yml

```
@@ -0,0 +1,226 @@
+{{- if .Subscribe}}
+SQSQueue:
+  Type: AWS::SQS::Queue
+  {{- if .Subscribe.Queue}}
+  Properties:
+    QueueName: !Sub ${AWS::StackName}-{{- if .Subscribe.Queue}}{{if .Subscribe.Queue.Name}}{{logicalIDSafe .Subscribe.Queue.Name}}{{- else}}SQSQueue{{- end}}{{- if .Subscribe.Queue.FIFO}}.fifo{{- end}}{{- end}}
```

Since `QueueName` is not a required field, what do you think of:
```suggestion
    {{- if.Subscribe.Queue.Name }}
    QueueName: {{.Subscribe.Queue.Name}}
    {{- end }}
```
i.e. we should use the name that's provided by the user if `name != ""` and if it's empty then let CloudFormation generate one for us.

---

This would mess with the ability to provide customers with a JSON of queue URIs, as we won't know what the names of the queues are.

---

Maybe I'm not understanding fully, but it looks like the queue names are conditional on the presence of `QueueName` either way, so queue URIs are going to vary either way, too, right? 

It would be nicer to use the provided name if possible.